File: ObjC_Example/Native/NativeAdPageContentViewController.m
//
//  NativeAdPageContentViewController.m
//  ObjC_Example
//
//  Copyright (c) 2015年 FAN Communications. All rights reserved.
//

#import "NativeAdPageContentViewController.h"

@interface NativeAdPageContentViewController ()

@property (nonatomic, weak) IBOutlet UIView<NADNativeViewRendering> *adView;
@property (nonatomic, weak) IBOutlet UILabel *label;

@end

@implementation NativeAdPageContentViewController

- (void)setAd:(NADNative *)ad
{
    [ad intoView:self.adView advertisingExplicitly:NADNativeAdvertisingExplicitlyPromotion];
}

- (void)setPosition:(NSUInteger)position
{
    self.label.text = [NSString stringWithFormat:@"Page%lu", (unsigned long)position];
}

@end

File: Swift_Example/Native/NativeAdPageContentViewController.swift
//
//  NativeAdPageContentViewController.swift
//  ObjC_Example
//
//  Copyright © 2015年 FAN Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdPageContentViewController: UIViewController {
    
    @IBOutlet fileprivate weak var adView: UIView!
    @IBOutlet fileprivate weak var label: UILabel!
    
    var ad: NADNative! {
        didSet {
            self.ad.intoView(self.adView as! (UIView & NADNativeViewRendering)?, advertisingExplicitly: .promotion)
        }
    }
    
    var position: Int! {
        didSet {
            self.label.text = "Page\(self.position!)"
        }
    }
}
